dace/runtime/include/dace/nan.h

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+#ifndef __DACE_NAN_H
+#define __DACE_NAN_H
+
+// Class to define a stateless NAN and related operators.
+
+namespace dace
+{
+    namespace math
+    {
+        //////////////////////////////////////////////////////
+        // Defines a typeless Pi
+        struct typeless_nan
+        {
+            operator int() const = delete;
+            operator float() const
+            {
+                return std::nanf("");
```

Comment:
`std::nan*` use `strto*`, which will be slow. Please use numeric_limits' NaN: https://en.cppreference.com/w/cpp/types/numeric_limits/quiet_NaN

Reply:
Thanks for reminding me about that thing.